### PR TITLE
Move QR scanner to popover window

### PR DESCRIPTION
**Pull request by jtormey**
@plondon we talked about adding something like this a few weeks ago, so I took a few minutes to put this together. All the functionality is there, but it might need some design help (in advanced send in particular).

- [x] fix QR processing in send when amount is present.
- [x] fix advanced send overflow issues (phil).
- [x] test for onAddressScan

qr-scan-popup

### CODE REVIEW

**Comment by Sjors** [comment]
Looks pretty sweet and I like how it removes complexity from the send controller and view.

Can you add a test for onAddressScan?

File: app/partials/send.jade [hunk 1]
```diff
@@ -51,8 +51,7 @@
                 name="{{ 'destinations' + $index }}"
                 ng-model="transaction.destinations[$index]"
                 ng-model-options="{ updateOn: 'blur' }"
-                ng-change="updateToLabel(); checkForSameDestination(); refreshTxProposal()"
-                on-request-qr="cameraOn($index)")
+                ng-change="updateToLabel(); checkForSameDestination(); refreshTxProposal()")
```
Comment (plondon) [inline app/partials/send.jade]: It doesn't look like we use refreshTxProposal() anywhere

File: app/partials/send.jade [hunk 2]
```diff
@@ -128,11 +127,6 @@
                          when="{'1': '{{block | translate}}', 'other': '{{blocks | translate}}' }")
             span ).
           span.has-error(ng-show="blockIdx > 6" translate="SMALL_FEE_WARNING")
-      //- QR Scanner 
-      .form-group.row(ng-hide="!cameraIsOn")
-        label.col-sm-2.col-xs-12(translate="QR")
-        .col-sm-10.col-xs-12
-          bc-qr-reader(active="cameraRequested", on-result="processURLfromQR" on-error="onError" camera-status="cameraIsOn")
```
Comment (plondon) [inline app/partials/send.jade]: removing processURLfromQR call also removes call to which removes the call to applyPaymentRequest, so if you scan a QR with an amount attached the amount field won't update.

**Comment by plondon** [comment]
looks awesome!

- added it to import-address as well and left a few comments re. processing the QR in send.
- the loading in and fading out are a little sketchy, but I can try and fix that.

**Comment by jtormey** [comment]
Great, thanks!

**Comment by Sjors** [comment]
If you to advanced send, add one row, scan an address QR code with the 2nd row, then scan the an address+amount QR code with the 1st row, the amount will be added to the 2nd row.

Also if you scan an address on the 2nd row and then scan another address on the 1st row, both rows will have the new address.

**Pull request merged**